# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../util-linux/dietlibc.patch
# Copyright (C) 2007 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./login-utils/vipw.c.orig	2007-07-17 13:07:32.000000000 -0400
+++ ./login-utils/vipw.c	2007-07-17 13:08:35.000000000 -0400
@@ -73,6 +73,10 @@
 
 #define FILENAMELEN 67
 
+#ifndef _PATH_VI
+#define _PATH_VI	"/usr/bin/vi"
+#endif
+
 char *progname;
 enum { VIPW, VIGR };
 int program;
--- ./login-utils/login.c.orig	2007-07-17 12:53:12.000000000 -0400
+++ ./login-utils/login.c	2007-07-17 12:53:35.000000000 -0400
@@ -152,7 +152,7 @@
 #ifndef __linux__
 #  include <tzfile.h>
 #endif
-#include <lastlog.h>
+#include <utmp.h>
 
 #define SLEEP_EXIT_TIMEOUT 5
 
--- ./mount/fstab.c.orig	2007-07-17 13:17:38.000000000 -0400
+++ ./mount/fstab.c	2007-07-17 13:17:59.000000000 -0400
@@ -399,9 +399,6 @@
 /* Flag to indicate that signals have been set up. */
 static int signals_have_been_setup = 0;
 
-/* Ensure that the lock is released if we are interrupted.  */
-extern char *strsignal(int sig);	/* not always in <string.h> */
-
 static void
 handler (int sig) {
      die(EX_USER, "%s", strsignal(sig));
--- ./sys-utils/dmesg.c.orig	2007-07-18 16:09:10.000000000 -0400
+++ ./sys-utils/dmesg.c	2007-07-18 16:10:01.000000000 -0400
@@ -35,7 +35,7 @@
 #include <stdlib.h>
 #include "nls.h"
 
-#if __GNU_LIBRARY__ < 5
+#if (__GNU_LIBRARY__ < 5) && (! defined(__dietlibc__))
 
 #ifndef __alpha__
 # define __NR_klogctl __NR_syslog
--- ./text-utils/column.c.orig	2007-07-18 16:24:58.000000000 -0400
+++ ./text-utils/column.c	2007-07-18 16:25:19.000000000 -0400
@@ -50,6 +50,7 @@
 #include "errs.h"
 #include "nls.h"
 
+#include <termios.h>
 #include "widechar.h"
 
 #ifdef ENABLE_WIDECHAR
--- ./partx/solaris.c.orig	2007-07-18 16:31:07.000000000 -0400
+++ ./partx/solaris.c	2007-07-18 16:31:37.000000000 -0400
@@ -1,12 +1,11 @@
 #include <stdio.h>
 #include <time.h>		/* time_t */
+#include <sys/types.h>		/* daddr_t */
 #include "partx.h"
 
 #define SOLARIS_X86_NUMSLICE	8
 #define SOLARIS_X86_VTOC_SANE	(0x600DDEEEUL)
 
-typedef int daddr_t;		/* or long - check */
-
 struct solaris_x86_slice {
 	unsigned short	s_tag;		/* ID tag of partition */
 	unsigned short	s_flag;		/* permision flags */
